pkg/template_generator.go

```diff
@@ -314,7 +384,16 @@ func addContainerValue(key string, s1 string, s2 string) string {
 }
 
 func addTemplateImageValue(containerName string, image string, key string, containerValue map[string]interface{}) string {
-	img := strings.Split(image, ":")
+	var img []string
+	var repo string = ""
+	if strings.Contains(image, "/") {
```

Comment:
@TamalSaha ,
1. As i see from test i run, 
```
image_str := strings.SplitN(image, "/", 2)
```
This condition solves the registry field problem if it finds "/". it will split only wrt to first "/".
Ex. for quay.io/coreos/flannel
repo: quay.io
image: coreos/flannel.

The resultant image name in values.yml will be "quay.io/coreos/flannel"
```
if len(repo) != 0 {
      containerValue[Image] = fmt.Sprintf("%s/%s", repo, imageName)
} 
```
Above line will populate values.yml correctly.

2. Regarding imageTag, the tag is extracted from already split image string, which does not contain registry URL ":" if present. So, it will split the correct tag. If it does not find any tag, the default is already set to "latest". So, i don't see any problem here.